### PR TITLE
[New Rule] Okta FastPass Phishing

### PR DESCRIPTION
**Pull request by austinsonger**
## Summary

This rule requires Okta to have the following turned on: 

Okta Identity Engine - select "Phishing Resistance for FastPass" under `Settings > Features` in the Admin Console.

If one or more users enrolled in FastPass is targeted using AiTM phishing kits, Okta Identity Engine identifies the failed origin check and generates a unique event in Okta System Log:

`eventType eq "user.authentication.auth_via_mfa" AND result eq "FAILURE" AND outcome.reason eq "FastPass declined phishing attempt"`

- https://sec.okta.com/fastpassphishingdetection 

`event.dataset:okta.system and event.category:authentication and 
  event.action:user.authentication.auth_via_mfa and okta.outcome.reason:result and okta.outcome.reason:"FastPass declined phishing attempt"`

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by austinsonger** [comment]
Why was this closed.

File: rules/integrations/okta/initial_access_okta_fastpass_phishing.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  okta.event_type:user.authentication.auth_via_mfa and event.outcome:failure and okta.outcome.reason:"FastPass declined phishing attempt"
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1078"
+name = "Valid Accounts"
+reference = "https://attack.mitre.org/techniques/T1078/"
```
Comment (terrancedejesus) [inline rules/integrations/okta/initial_access_okta_fastpass_phishing.toml:45]:
```suggestion
id = "T1078"
name = "Valid Accounts"
reference = "https://attack.mitre.org/techniques/T1078/"
```

id = "T1566"
name = "Phishing"
reference = "https://attack.mitre.org/techniques/T1566/"

File: rules/integrations/okta/initial_access_okta_fastpass_phishing.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+[metadata]
+creation_date = "2023/05/07"
+integration = ["okta"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/05/07"
```
Comment (terrancedejesus) [inline rules/integrations/okta/initial_access_okta_fastpass_phishing.toml:7]:
```suggestion
updated_date = "2023/11/07"
```

File: rules/integrations/okta/initial_access_okta_fastpass_phishing.toml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+The Okta Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://developer.okta.com/docs/reference/api/system-log/",
+    "https://developer.okta.com/docs/reference/api/event-types/",
+    "https://sec.okta.com/fastpassphishingdetection",
+]
+risk_score = 47
+rule_id = "ee39a9f7-5a79-4b0a-9815-d36b3cf28d3e"
+severity = "medium"
+tags = ["Elastic", "Identity", "Okta", "Continuous Monitoring", "SecOps", "Identity and Access","Initial Access"]
```
Comment (terrancedejesus) [inline rules/integrations/okta/initial_access_okta_fastpass_phishing.toml:30]:
```suggestion
tags = ["Tactic: Initial Access", "Use Case: Identity and Access Audit", "Data Source: Okta"]
```

File: rules/integrations/okta/initial_access_okta_fastpass_phishing.toml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+language = "kuery"
+license = "Elastic License v2"
+name = "Okta FastPass Phishing Detection"
+note = """## Setup
+
+The Okta Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://developer.okta.com/docs/reference/api/system-log/",
+    "https://developer.okta.com/docs/reference/api/event-types/",
+    "https://sec.okta.com/fastpassphishingdetection",
```
Comment (terrancedejesus) [inline rules/integrations/okta/initial_access_okta_fastpass_phishing.toml:25]:
```suggestion
    "https://sec.okta.com/fastpassphishingdetection",
    "https://help.okta.com/oie/en-us/content/topics/security/behavior-detection/about-behavior-detection.htm"
```

**Comment by terrancedejesus** [comment]
@austinsonger - Apologies for the oversight and delay in getting this rule in! Ive made some suggestions to update the rule based on recent repository requirements. Once adjusted, I'll approve. Thanks as always for being your contributions!

File: rules/integrations/okta/initial_access_okta_fastpass_phishing.toml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+Detects when Okta FastPass prevents a user from authenticating to a phishing website.
+"""
+
+index = ["filebeat-*", "logs-okta*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Okta FastPass Phishing Detection"
+note = """## Setup
+
+The Okta Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
```
Comment (Samirbous) [inline rules/integrations/okta/initial_access_okta_fastpass_phishing.toml:21]:
```suggestion
note = """## Setup

The Okta Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule.

This rule requires Okta to have the following turned on:

Okta Identity Engine - select 'Phishing Resistance for FastPass' under Settings > Features in the Admin Console.
"""
```

**Review by Samirbous — APPROVED** [review]

**Pull request merged**